pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -450,16 +453,44 @@ func (gce *GCECloud) EnsureTCPLoadBalancer(name, region string, loadBalancerIP n
 	return status, nil
 }
 
-// This is kind of hacky, but the managed instance group adds 4 random chars and a hyphen
-// to the base name. Older naming schemes put a hyphen and an incrementing index after
-// the base name. Thus we pull off the characters after the final dash to support both.
-func (gce *GCECloud) computeHostTag(host string) string {
-	host = strings.SplitN(host, ".", 2)[0]
-	lastHyphen := strings.LastIndex(host, "-")
-	if lastHyphen == -1 {
-		return host
+// We grab all tags from all instances being added to the pool.
+// * The longest tag that is a prefix of the instance name is used
+// * If any instance has a prefix tag, all instances must
+// * If no instances have a prefix tag, no tags are used
+func (gce *GCECloud) computeHostTags(hosts []string) ([]string, error) {
```

Comment:
See  #14430, only in the current GKE case can you assume that all instances will be tagged the same.  This allows for setting separate tags per instance and when the pool logic only adds the instances serving the pods to the pool, you will end up with only those instances allowed to receive that traffic.